patchwork-data-generators/src/main/java/net/minecraftforge/fml/event/lifecycle/GatherDataEvent.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Minecraft Forge, Patchwork Project
+ * Copyright (c) 2016-2020, 2019-2020
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.fml.event.lifecycle;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.function.Function;
+
+import net.minecraftforge.client.model.generators.ExistingFileHelper;
+import net.minecraftforge.fml.ModContainer;
+
+import net.minecraft.data.DataGenerator;
+
+public class GatherDataEvent extends ModLifecycleEvent {
+	private final DataGenerator dataGenerator;
+	private final DataGeneratorConfig config;
+	private final ExistingFileHelper existingFileHelper;
+
+	public GatherDataEvent(final ModContainer modContainer, final DataGenerator dataGenerator, final DataGeneratorConfig dataGeneratorConfig, ExistingFileHelper existingFileHelper) {
+		super(modContainer);
+		this.dataGenerator = dataGenerator;
+		this.config = dataGeneratorConfig;
+		this.existingFileHelper = existingFileHelper;
+	}
+
+	public DataGenerator getGenerator() {
+		return this.dataGenerator;
+	}
+
+	public ExistingFileHelper getExistingFileHelper() {
+		return existingFileHelper;
+	}
+
+	public boolean includeServer() {
+		return this.config.server;
+	}
+
+	public boolean includeClient() {
+		return this.config.client;
+	}
+
+	public boolean includeDev() {
+		return this.config.dev;
+	}
+
+	public boolean includeReports() {
+		return this.config.reports;
+	}
+
+	public boolean validate() {
+		return this.config.validate;
+	}
+
+	public static class DataGeneratorConfig {
+		private final Set<String> mods;
+		private final Path path;
+		private final Collection<Path> inputs;
+		private final boolean server;
+		private final boolean client;
+		private final boolean dev;
+		private final boolean reports;
+		private final boolean validate;
+		private List<DataGenerator> generators = new ArrayList<>();
```

Comment:
This field can be final too
```suggestion
		private final List<DataGenerator> generators = new ArrayList<>();
```